src/js/components/user/user.js

```diff
@@ -78,45 +85,56 @@ class User extends MaharaBaseComponent {
 
           let lang = this.props.lang.find(l => langOptions.indexOf(l) > -1);
 
-        return <ReactPullToRefresh onRefresh={maharaServer.refreshUserProfile} hammerOptions={{'touchAction': 'auto'}}>
-                <section>
-                  <h2>{siteName}</h2>
-                  <div className="userBlock">
-                    <img src={icon} className="profile"/>
-                    <div style={{'marginRight': '1em'}}>{displayName}</div>
-                    <a onClick={this.logoutButton} className="logout">{this.gettext("logout_button") }</a>
-                  </div>
-                  <div className="userInfoBlock">
-                    {this.renderServer() }
-
-                    <MaharaSelector
-                      label={this.gettext("default_journal")}
-                      defaultOption={defaultJournal}
-                      name="journalSelect"
-                      options={journalOptions}
-                      onSetSelection={this.setJournal}
-                      ></MaharaSelector>
-
-                    <MaharaSelector
-                      label={this.gettext("default_language")}
-                      defaultOption={{'text': lang, 'id': lang}}
-                      name="languageSelect"
-                      options={langOptions}
-                      onSetSelection={this.setLanguage}
-                      ></MaharaSelector>
+        if (this.state.settingsPage) {
```

Comment:
the if/else is a bit gross, perhaps a better pattern would be a parent component with any shared functionality there, and subclass it for the two different flavours?